modules/storages/app/common/storages/peripherals/connection_validators/nextcloud/base_validator.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Peripherals
+    module ConnectionValidators
+      module Nextcloud
+        class BaseValidator
+          def initialize(storage)
+            @storage = storage
+          end
+
+          def call
+            catch :interrupted do
+              validate
+            end
+
+            @results
+          end
+
+          private
+
+          def validate = raise Errors::SubclassResponsibility
+
+          def register_checks(*keys)
+            @results = keys.to_h { [it, CheckResult.skipped(it)] }
```

Comment:
:green_circle: The current implementation of this implies that I can only call `register_checks` once, never twice.
 
I think that's enough, but it limits the ways in which I can write my validator a bit. E.g. this layout would not work:

```ruby
def validate
  register_checks(:common_check)
  if rand < 0.5
    register_checks(*a_checks)
    validate_a
  else
    register_checks(*b_checks)
    validate_b
  end
end
```

The other issue is that if I for whatever reason don't add any tests, we'd now return `nil` instead of returning `{}`. As indicated by the green traffic light, I think both of these problems are absolutely minor, but I wanted to explicitly write them down at least.